File: 1013-fibonacci-number/fibonacci-number.py
class Solution:
    def fib(self, n: int) -> int:
        if n<=1:
            return 0 if n==0 else 1
        dp = [1] * (n+1)
        dp[1] = 1
        for i in range(2,n+1):
            dp[i] = dp[i-1] + dp[i-2]
        return dp[n-1]

File: 1013-fibonacci-number/fibonacci-number.java
class Solution {
    Map<Integer,Integer> map = new HashMap<>();
    public int fib(int n) {
        if(map.containsKey(n))return map.get(n);
        if(n==0)return 0;
        if(n==1||n==2)return 1;
        int res = fib(n-1)+fib(n-2);
        map.put(n,res);
        return res;
    }
}